pkg/sdk/parameters_impl.go

```diff
@@ -20,115 +20,88 @@ func GetSessionParametersFrom(params map[string]any) (*SessionParameters, error)
+		err = setBooleanValue(parameter, value, &sessionParameters.ClientResultColumnCaseInsensitive)
+	case SessionParameterClientMetadataUseSessionDatabase:
+		err = setBooleanValue(parameter, value, &sessionParameters.ClientMetadataUseSessionDatabase)
+	case SessionParameterClientSessionKeepAlive:
+		err = setBooleanValue(parameter, value, &sessionParameters.ClientSessionKeepAlive)
+	case SessionParameterClientSessionKeepAliveHeartbeatFrequency:
+		err = setIntegerValue(parameter, value, &sessionParameters.ClientSessionKeepAliveHeartbeatFrequency)
+	case SessionParameterClientTimestampTypeMapping:
+		sessionParameters.ClientTimestampTypeMapping = Pointer(ClientTimestampTypeMapping(value))
 	case SessionParameterDateInputFormat:
 		sessionParameters.DateInputFormat = &value
 	case SessionParameterDateOutputFormat:
 		sessionParameters.DateOutputFormat = &value
+	case SessionParameterEnableUnloadPhysicalTypeOptimization:
+		err = setBooleanValue(parameter, value, &sessionParameters.EnableUnloadPhysicalTypeOptimization)
 	case SessionParameterErrorOnNondeterministicMerge:
-		b, err := parseBooleanParameter(string(parameter), value)
-		if err != nil {
-			return err
-		}
-		sessionParameters.ErrorOnNondeterministicMerge = b
+		err = setBooleanValue(parameter, value, &sessionParameters.ErrorOnNondeterministicMerge)
 	case SessionParameterErrorOnNondeterministicUpdate:
-		b, err := parseBooleanParameter(string(parameter), value)
-		if err != nil {
-			return err
-		}
-		sessionParameters.ErrorOnNondeterministicUpdate = b
+		err = setBooleanValue(parameter, value, &sessionParameters.ErrorOnNondeterministicUpdate)
 	case SessionParameterGeographyOutputFormat:
 		sessionParameters.GeographyOutputFormat = Pointer(GeographyOutputFormat(value))
+	case SessionParameterGeometryOutputFormat:
+		sessionParameters.GeometryOutputFormat = Pointer(GeometryOutputFormat(value))
+	case SessionParameterJdbcTreatDecimalAsInt:
+		err = setBooleanValue(parameter, value, &sessionParameters.JdbcTreatDecimalAsInt)
+	case SessionParameterJdbcTreatTimestampNtzAsUtc:
+		err = setBooleanValue(parameter, value, &sessionParameters.JdbcTreatTimestampNtzAsUtc)
+	case SessionParameterJdbcUseSessionTimezone:
+		err = setBooleanValue(parameter, value, &sessionParameters.JdbcUseSessionTimezone)
 	case SessionParameterJSONIndent:
-		v, err := strconv.Atoi(value)
-		if err != nil {
-			return fmt.Errorf("JSON_INDENT session parameter is an integer, got %v", value)
-		}
-		sessionParameters.JSONIndent = Pointer(v)
+		err = setIntegerValue(parameter, value, &sessionParameters.JSONIndent)
 	case SessionParameterLockTimeout:
-		v, err := strconv.Atoi(value)
-		if err != nil {
-			return fmt.Errorf("LOCK_TIMEOUT session parameter is an integer, got %v", value)
-		}
-		sessionParameters.LockTimeout = Pointer(v)
+		err = setIntegerValue(parameter, value, &sessionParameters.LockTimeout)
+	case SessionParameterLogLevel:
+		sessionParameters.LogLevel = Pointer(LogLevel(value))
```

Comment:
For enum do we want to use ToX(...) functions for enums? If yes, then ToLogLevel is available.

Reply:
I was thinking about this and was worried about possible breaking changes. Using a bad value should end with an error either way, but sometimes, values that we do not accept are accepted in Snowflake (so this is a part I was worried about). These parameters are used in multiple resources, so it's a bit of a waste to now try to announce this potential change in each of them. We can for sure do it with the redesign, because we are listing the possible values then.

Would you be okay with the TODO for session_parameters issue?

Reply:
Yeah, the TODO would be ok. I was thinking about something similar to a configuration list (just a slice of structs) that would generate appropriate mappings (for a future improvement ofc) etc where it could look similar to
```go
var parameters = {
  "param_name": {Levels: []Level{Account, Session, User}, Type: TypeInt, ....}
}
```